src/main/java/org/folio/edge/rtac/RtacHandler.java

```diff
@@ -1,114 +1,195 @@
-
-        RtacOkapiClient rtacClient = new RtacOkapiClient(client);
-        String instanceIds;
-        try {
-          List<String> ids = getListOfIds(isBatch, params);
-          if (CollectionUtils.isNullOrEmpty(ids)) {
-            badRequest(ctx, "Invalid instance id" + params.toString());
+        new String[]{},
+        new String[]{PARAM_TITLE_ID, PARAM_INSTANCE_ID, PARAM_INSTANCE_IDS, PARAM_FULL_PERIODICALS},
+        (client, params) -> {
+
+          RtacOkapiClient rtacClient = new RtacOkapiClient(client);
+          String instanceIds;
+          try {
+            List<String> ids = getListOfIds(isBatch, params);
+            if (CollectionUtils.isEmpty(ids)) {
+              badRequest(ctx, "Invalid instance id" + params.toString());
+              return;
+            }
+
+            var rtacParams = new HashMap<>();
+            rtacParams.put(PARAM_INSTANCE_IDS, ids);
+            rtacParams
+                .put(PARAM_FULL_PERIODICALS, Boolean.valueOf(params.get(PARAM_FULL_PERIODICALS)));
+            instanceIds = Mappers.jsonMapper.writeValueAsString(rtacParams);
+          } catch (JsonProcessingException e) {
+            logger.error("Exception during serialization in mod-rtac", e);
+            returnEmptyResponse(ctx);
             return;
           }
 
-          var rtacParams = new HashMap<>();
-          rtacParams.put(PARAM_INSTANCE_IDS, ids);
-          rtacParams.put(PARAM_FULL_PERIODICALS, Boolean.valueOf(params.get(PARAM_FULL_PERIODICALS)));
-          instanceIds = Mappers.jsonMapper.writeValueAsString(rtacParams);
-        } catch (JsonProcessingException e) {
-          logger.error("Exception during serialization in mod-rtac", e);
-          returnEmptyResponse(ctx);
-          return;
-        }
-        rtacClient.rtac(instanceIds, request.headers())
-          .thenAcceptAsync(body -> {
-            try {
-              logger.debug("rtac response: {}", body);
-              String xml;
-              final var instances = Instances.fromJson(body);
-              if (isBatch) {
-                xml = instances.toXml();
-              } else {
-                var holdings = instances.getHoldings().isEmpty() ? new Holdings() : instances.getHoldings().get(0);
-                xml = holdings.toXml();
-              }
-              logger.info("Converted Response: \n {}", xml);
-              ctx.response()
-                .setStatusCode(200)
-                .putHeader(HttpHeaders.CONTENT_TYPE, APPLICATION_XML)
-                .end(xml);
-            } catch (IOException e) {
-              logger.error("Exception translating JSON -> XML: {}", e.getMessage(), e);
-              returnEmptyResponse(ctx);
-            }
-          })
-          .exceptionally(t -> {
-            logger.error("Exception calling mod-rtac", t);
-            returnEmptyResponse(ctx);
-            return null;
-          });
+          // Remember original request types
+          saveRequestType(request);
```

Comment:
RE: my previous comment, why not do something like:
```suggestion
          final boolean isXmlResponse = isXmlRequest(request);
```
instead of storing the accept data. Then we just remove the Accept header from `request.headers()` before calling the rtac client.

Reply:
@mreno-EBSCO  I agree with this, it's where I was going with the conversation about forwarding headers

Reply:
I added some extra processing to do not change any data received from client, even headers, therefore the main point was to hide non-json type info and then restore in back. I understand that possibly data from accept header is only for server processing, but it's possibly not a best approach, I mean changing of headers. Am I understand you correctly that you was talking about just replacing accept header data before passing request to RTAC without restoring back?

Reply:
I guess my main question is why do we need to restore the Accept header? It is a request header, I don't think the response needs to contain an Accept header. What would that mean to the client? I don't think we need to set this header when calling mod-rtac since I believe it would default to all content type if not set.

Reply:
>. I don't think we need to set this header when calling mod-rtac since I believe it would default to all content type if not set

I'm not sure many / any of the internal modules in FOLIO actually respect the accept header

Reply:
You are right, it's a request header and we don't return it back. What I mean that it could be used for other processing in the code at all on the server side, beside this handle method. That's why I restored it. Taking into account similar question above it looks it would be better to simplify the process of replacement to make it as easy as it can be.